<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: P-values and test-statistics from the Hedenfalk et al. (2001)...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for hedenfalk {qvalue}"><tr><td>hedenfalk {qvalue}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>P-values and test-statistics from the Hedenfalk et al. (2001) gene
expression dataset</h2>

<h3>Description</h3>

<p>The data from the breast cancer gene expression study of
Hedenfalk et al. (2001) were obtained and analyzed. A comparison was made
between 3,226 genes of two mutation types, BRCA1 (7 arrays) and BRCA2 (8
arrays). The data included here are p-values, test-statistics, and
permutation null test-statistics obtained from a two-sample t-test analysis
on a set of 3170 genes, as described in Storey and Tibshirani (2003).
</p>


<h3>Usage</h3>

<pre>
data(hedenfalk)
</pre>


<h3>Value</h3>

<p>A list called <code>hendfalk</code> containing:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>p</code></td>
<td>
<p>Vector of 3,170
p-values of tests comparing BRCA1 to BRCA2.</p>
</td></tr>
<tr valign="top"><td><code>stat</code></td>
<td>
<p>Vector of 3,170
absolute two-sample t-statistics comparing BRCA1 to BRCA2.</p>
</td></tr>
<tr valign="top"><td><code>stat0</code></td>
<td>
<p>A
3,170 by 100 matrix of absolute two-sample t-statistics from 100 independent
permutations of the BRCA1 and BRCA2 labels; the row <code>stat0[i,].</code>
contains the permutation statistics corresponding to observed statistic
<code>stat[i]</code>.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Hedenfalk I et al. (2001). Gene expression profiles in hereditary breast cancer.
New England Journal of Medicine, 344: 539-548.
</p>
<p>Storey JD and Tibshirani R. (2003). Statistical significance for genome-wide
studies. Proceedings of the National Academy of Sciences, 100: 9440-9445. <br />
<a href="http://www.pnas.org/content/100/16/9440.full">http://www.pnas.org/content/100/16/9440.full</a>
</p>


<h3>See Also</h3>

<p><code><a href="../../qvalue/help/qvalue.html">qvalue</a></code>, <code><a href="../../qvalue/help/empPvals.html">empPvals</a></code>
</p>


<h3>Examples</h3>

<pre>
# import data
data(hedenfalk)
stat &lt;- hedenfalk$stat
stat0 &lt;- hedenfalk$stat0 #vector from null distribution

p.pooled &lt;- empPvals(stat=stat, stat0=stat0)
p.testspecific &lt;- empPvals(stat=stat, stat0=stat0, pool=FALSE)

#compare pooled to test-specific p-values
qqplot(p.pooled, p.testspecific); abline(0,1)

# calculate q-values and view results
qobj &lt;- qvalue(p.pooled)
summary(qobj)
hist(qobj)
plot(qobj)

</pre>

<hr /><div style="text-align: center;">[Package <em>qvalue</em> version 2.26.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
