<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Perform principal component analysis</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for pca {pcaMethods}"><tr><td>pca {pcaMethods}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Perform principal component analysis</h2>

<h3>Description</h3>

<p>Perform PCA on a numeric matrix for visualisation, information
extraction and missing value imputation.
</p>


<h3>Usage</h3>

<pre>
pca(object, method, nPcs = 2, scale = c("none", "pareto", "vector",
  "uv"), center = TRUE, completeObs = TRUE, subset = NULL,
  cv = c("none", "q2"), ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>object</code></td>
<td>
<p>Numerical matrix with (or an object coercible to
such) with samples in rows and variables as columns. Also takes
<code>ExpressionSet</code> in which case the transposed expression
matrix is used. Can also be a data frame in which case all
numberic variables are used to fit the PCA.</p>
</td></tr>
<tr valign="top"><td><code>method</code></td>
<td>
<p>One of the methods reported by
<code>listPcaMethods()</code>. Can be left missing in which case the
<code>svd</code> PCA is chosen for data wihout missing values and
<code>nipalsPca</code> for data with missing values</p>
</td></tr>
<tr valign="top"><td><code>nPcs</code></td>
<td>
<p>Number of principal components to calculate.</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>Scaling, see <code><a href="../../pcaMethods/help/prep.html">prep</a></code>.</p>
</td></tr>
<tr valign="top"><td><code>center</code></td>
<td>
<p>Centering, see <code><a href="../../pcaMethods/help/prep.html">prep</a></code>.</p>
</td></tr>
<tr valign="top"><td><code>completeObs</code></td>
<td>
<p>Sets the <code>completeObs</code> slot on the
resulting <code>pcaRes</code> object containing the original data with
but with all NAs replaced with the estimates.</p>
</td></tr>
<tr valign="top"><td><code>subset</code></td>
<td>
<p>A subset of variables to use for calculating the
model. Can be column names or indices.</p>
</td></tr>
<tr valign="top"><td><code>cv</code></td>
<td>
<p>character naming a the type of cross-validation
to be performed.</p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>Arguments to <code><a href="../../pcaMethods/help/prep.html">prep</a></code>, the chosen pca
method and <code><a href="../../pcaMethods/help/Q2.html">Q2</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method is wrapper function for the following set of pca
methods:
</p>
<dl>
<dt>svd:</dt><dd><p>Uses classical <code>prcomp</code>. See
documentation for <code><a href="../../pcaMethods/help/svdPca.html">svdPca</a></code>.</p>
</dd>
<dt>nipals:</dt><dd><p>An iterative method capable of handling small
amounts of missing values. See documentation for
<code><a href="../../pcaMethods/help/nipalsPca.html">nipalsPca</a></code>.</p>
</dd>
<dt>rnipals:</dt><dd><p>Same as nipals but implemented in R.</p>
</dd>
<dt>bpca:</dt><dd><p>An iterative method using a Bayesian model to handle
missing values. See documentation for <code><a href="../../pcaMethods/help/bpca.html">bpca</a></code>.</p>
</dd>
<dt>ppca:</dt><dd><p>An iterative method using a probabilistic model to
handle missing values. See documentation for <code><a href="../../pcaMethods/help/ppca.html">ppca</a></code>.</p>
</dd>
<dt>svdImpute:</dt><dd><p>Uses expectation maximation to perform SVD PCA
on incomplete data. See documentation for
<code><a href="../../pcaMethods/help/svdImpute.html">svdImpute</a></code>.</p>
</dd></dl>

<p>Scaling and centering is part of the PCA model and handled by
<code><a href="../../pcaMethods/help/prep.html">prep</a></code>.
</p>


<h3>Value</h3>

<p>A <code>pcaRes</code> object.
</p>


<h3>Author(s)</h3>

<p>Wolfram Stacklies, Henning Redestig
</p>


<h3>References</h3>

<p>Wold, H. (1966) Estimation of principal components and
related models by iterative least squares. In Multivariate
Analysis (Ed., P.R. Krishnaiah), Academic Press, NY, 391-420.
</p>
<p>Shigeyuki Oba, Masa-aki Sato, Ichiro Takemasa, Morito Monden,
Ken-ichi Matsubara and Shin Ishii.  A Bayesian missing value
estimation method for gene expression profile
data. <em>Bioinformatics, 19(16):2088-2096, Nov 2003</em>.
</p>
<p>Troyanskaya O. and Cantor M. and Sherlock G. and Brown P. and
Hastie T. and Tibshirani R. and Botstein D. and Altman RB.  -
Missing value estimation methods for DNA microarrays.
<em>Bioinformatics. 2001 Jun;17(6):520-5</em>.
</p>


<h3>See Also</h3>

<p><code><a href="../../stats/html/prcomp.html">prcomp</a></code>, <code><a href="../../stats/html/princomp.html">princomp</a></code>,
<code><a href="../../pcaMethods/help/nipalsPca.html">nipalsPca</a></code>, <code><a href="../../pcaMethods/help/svdPca.html">svdPca</a></code>
</p>


<h3>Examples</h3>

<pre>
data(iris)
##  Usually some kind of scaling is appropriate
pcIr &lt;- pca(iris, method="svd", nPcs=2)
pcIr &lt;- pca(iris, method="nipals", nPcs=3, cv="q2")
## Get a short summary on the calculated model
summary(pcIr)
plot(pcIr)
## Scores and loadings plot
slplot(pcIr, sl=as.character(iris[,5]))

## use an expressionset and ggplot
data(sample.ExpressionSet)
pc &lt;- pca(sample.ExpressionSet)
df &lt;- merge(scores(pc), pData(sample.ExpressionSet), by=0)
library(ggplot2)
ggplot(df, aes(PC1, PC2, shape=sex, color=type)) +
  geom_point() +
  xlab(paste("PC1", pc@R2[1] * 100, "% of the variance")) +
  ylab(paste("PC2", pc@R2[2] * 100, "% of the variance"))
</pre>

<hr /><div style="text-align: center;">[Package <em>pcaMethods</em> version 1.86.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
