<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Gene Ontology-based Sematic Similarity Measures</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for GOSemSim-package {GOSemSim}"><tr><td>GOSemSim-package {GOSemSim}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Gene Ontology-based Sematic Similarity Measures</h2>

<h3>Description</h3>

<p>Implementation of semantic similarity measures to estimate the functional
similarities among Gene Ontology terms and gene products
</p>


<h3>Details</h3>

<p>Quantitative measure of functional similarities among gene products is
important for post-genomics study. and widely used in gene function
prediction, cluster analysis and pathway modeling.  This package is designed
to estimate the GO terms' and genes' semantic similarities.  Implemented five
methods proposed by Resnik, Schlicker, Jiang, Lin and Wang respectively.
Support many species, including Anopheles, Arabidopsis, Bovine, Canine,
Chicken, Chimp, E coli strain K12 and strain Sakai, Fly, Human, Malaria,
Mouse, Pig, Rhesus, Rat, Worm, Xenopus, Yeast, Zebrafish.
</p>

<table summary="Rd table">
<tr>
 <td style="text-align: left;"> Package: </td><td style="text-align: left;"> GOSemSim</td>
</tr>
<tr>
 <td style="text-align: left;"> Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;"> Version: </td><td style="text-align: left;">
2.0.0</td>
</tr>
<tr>
 <td style="text-align: left;"> Date: </td><td style="text-align: left;"> 09-11-2012</td>
</tr>
<tr>
 <td style="text-align: left;"> biocViews:</td><td style="text-align: left;"> GO, Clustering, Pathways,
Anopheles_gambiae, Arabidopsis_thaliana, Bos_taurus, Caenorhabditis_elegans,
Canis_familiaris, Danio_rerio, Drosophila_melanogaster, Escherichia_coli,
Gallus_gallus, Homo_sapiens, Mus_musculus, Pan_troglodytes,
Plasmodium_falciparum, Rattus_norvegicus, Saccharomyces_cerevisiae,
Streptomyces_coelicolor, Sus_scrofa, Xenopus_laevis</td>
</tr>
<tr>
 <td style="text-align: left;"> Depends:</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> methods, AnnotationDbi, GO.db</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests:</td><td style="text-align: left;"> clusterProfiler, DOSE</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> Artistic-2.0</td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>
<p>Maintainer: Guangchuang Yu &lt;guangchuangyu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Yu et al. (2010) GOSemSim: an R package for measuring semantic
similarity among GO terms and gene products <em>Bioinformatics</em> (Oxford,
England), 26:7 976&ndash;978, April 2010. ISSN 1367-4803
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976">http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976</a>
PMID: 20179076
</p>


<h3>See Also</h3>

<p><code><a href="../../GOSemSim/help/goSim.html">goSim</a></code> <code><a href="../../GOSemSim/help/mgoSim.html">mgoSim</a></code> <code><a href="../../GOSemSim/help/geneSim.html">geneSim</a></code>
<code><a href="../../GOSemSim/help/mgeneSim.html">mgeneSim</a></code> <code><a href="../../GOSemSim/help/clusterSim.html">clusterSim</a></code> <code><a href="../../GOSemSim/help/mclusterSim.html">mclusterSim</a></code>
</p>

<hr /><div style="text-align: center;">[Package <em>GOSemSim</em> version 2.20.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
