<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head><title>R: Differential Enrichment analysis of Proteomics data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">
<h1> Differential Enrichment analysis of Proteomics data
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;DEP&rsquo; version 1.16.0</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">User guides, package vignettes and other documentation.</a></li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<table width="100%">
<tr><td style="width: 25%;"><a href="DEP.html">DEP-package</a></td>
<td>DEP: A package for Differential Enrichment analysis of Proteomics data.</td></tr>
<tr><td style="width: 25%;"><a href="add_rejections.html">add_rejections</a></td>
<td>Mark significant proteins</td></tr>
<tr><td style="width: 25%;"><a href="analyze_dep.html">analyze_dep</a></td>
<td>Differential expression analysis</td></tr>
<tr><td style="width: 25%;"><a href="DEP.html">DEP</a></td>
<td>DEP: A package for Differential Enrichment analysis of Proteomics data.</td></tr>
<tr><td style="width: 25%;"><a href="DiUbi.html">DiUbi</a></td>
<td>DiUbi - Ubiquitin interactors for different diubiquitin-linkages (UbIA-MS dataset)</td></tr>
<tr><td style="width: 25%;"><a href="DiUbi_ExpDesign.html">DiUbi_ExpDesign</a></td>
<td>Experimental design of the DiUbi dataset</td></tr>
<tr><td style="width: 25%;"><a href="filter_missval.html">filter_missval</a></td>
<td>Filter on missing values</td></tr>
<tr><td style="width: 25%;"><a href="filter_proteins.html">filter_proteins</a></td>
<td>Filter proteins based on missing values</td></tr>
<tr><td style="width: 25%;"><a href="get_df_long.html">get_df_long</a></td>
<td>Generate a long data.frame from a SummarizedExperiment</td></tr>
<tr><td style="width: 25%;"><a href="get_df_wide.html">get_df_wide</a></td>
<td>Generate a wide data.frame from a SummarizedExperiment</td></tr>
<tr><td style="width: 25%;"><a href="get_prefix.html">get_prefix</a></td>
<td>Obtain the longest common prefix</td></tr>
<tr><td style="width: 25%;"><a href="get_results.html">get_results</a></td>
<td>Generate a results table</td></tr>
<tr><td style="width: 25%;"><a href="get_suffix.html">get_suffix</a></td>
<td>Obtain the longest common suffix</td></tr>
<tr><td style="width: 25%;"><a href="import_IsobarQuant.html">import_IsobarQuant</a></td>
<td>Import from IsobarQuant</td></tr>
<tr><td style="width: 25%;"><a href="import_MaxQuant.html">import_MaxQuant</a></td>
<td>Import from MaxQuant</td></tr>
<tr><td style="width: 25%;"><a href="impute.html">impute</a></td>
<td>Impute missing values</td></tr>
<tr><td style="width: 25%;"><a href="LFQ.html">LFQ</a></td>
<td>LFQ workflow</td></tr>
<tr><td style="width: 25%;"><a href="make_se.html">make_se</a></td>
<td>Data.frame to SummarizedExperiment object conversion using an experimental design</td></tr>
<tr><td style="width: 25%;"><a href="make_se_parse.html">make_se_parse</a></td>
<td>Data.frame to SummarizedExperiment object conversion using parsing from column names</td></tr>
<tr><td style="width: 25%;"><a href="make_unique.html">make_unique</a></td>
<td>Make unique names</td></tr>
<tr><td style="width: 25%;"><a href="manual_impute.html">manual_impute</a></td>
<td>Imputation by random draws from a manually defined distribution</td></tr>
<tr><td style="width: 25%;"><a href="meanSdPlot.html">meanSdPlot</a></td>
<td>Plot row standard deviations versus row means</td></tr>
<tr><td style="width: 25%;"><a href="normalize_vsn.html">normalize_vsn</a></td>
<td>Normalization using vsn</td></tr>
<tr><td style="width: 25%;"><a href="plot_all.html">plot_all</a></td>
<td>Visualize the results in different types of plots</td></tr>
<tr><td style="width: 25%;"><a href="plot_cond.html">plot_cond</a></td>
<td>Plot frequency of significant conditions per protein and the overlap in proteins between conditions</td></tr>
<tr><td style="width: 25%;"><a href="plot_cond_freq.html">plot_cond_freq</a></td>
<td>Plot frequency of significant conditions per protein</td></tr>
<tr><td style="width: 25%;"><a href="plot_cond_overlap.html">plot_cond_overlap</a></td>
<td>Plot conditions overlap</td></tr>
<tr><td style="width: 25%;"><a href="plot_cor.html">plot_cor</a></td>
<td>Plot correlation matrix</td></tr>
<tr><td style="width: 25%;"><a href="plot_coverage.html">plot_coverage</a></td>
<td>Plot protein coverage</td></tr>
<tr><td style="width: 25%;"><a href="plot_detect.html">plot_detect</a></td>
<td>Visualize intensities of proteins with missing values</td></tr>
<tr><td style="width: 25%;"><a href="plot_dist.html">plot_dist</a></td>
<td>Plot Gower's distance matrix</td></tr>
<tr><td style="width: 25%;"><a href="plot_frequency.html">plot_frequency</a></td>
<td>Plot protein overlap between samples</td></tr>
<tr><td style="width: 25%;"><a href="plot_gsea.html">plot_gsea</a></td>
<td>Plot enriched Gene Sets</td></tr>
<tr><td style="width: 25%;"><a href="plot_heatmap.html">plot_heatmap</a></td>
<td>Plot a heatmap</td></tr>
<tr><td style="width: 25%;"><a href="plot_imputation.html">plot_imputation</a></td>
<td>Visualize imputation</td></tr>
<tr><td style="width: 25%;"><a href="plot_missval.html">plot_missval</a></td>
<td>Plot a heatmap of proteins with missing values</td></tr>
<tr><td style="width: 25%;"><a href="plot_normalization.html">plot_normalization</a></td>
<td>Visualize normalization</td></tr>
<tr><td style="width: 25%;"><a href="plot_numbers.html">plot_numbers</a></td>
<td>Plot protein numbers</td></tr>
<tr><td style="width: 25%;"><a href="plot_pca.html">plot_pca</a></td>
<td>Plot PCA</td></tr>
<tr><td style="width: 25%;"><a href="plot_p_hist.html">plot_p_hist</a></td>
<td>Plot a P value histogram</td></tr>
<tr><td style="width: 25%;"><a href="plot_single.html">plot_single</a></td>
<td>Plot values for a protein of interest</td></tr>
<tr><td style="width: 25%;"><a href="plot_volcano.html">plot_volcano</a></td>
<td>Volcano plot</td></tr>
<tr><td style="width: 25%;"><a href="process.html">process</a></td>
<td>Proteomics data processing</td></tr>
<tr><td style="width: 25%;"><a href="report.html">report</a></td>
<td>Generate a markdown report</td></tr>
<tr><td style="width: 25%;"><a href="run_app.html">run_app</a></td>
<td>DEP shiny apps</td></tr>
<tr><td style="width: 25%;"><a href="se2msn.html">se2msn</a></td>
<td>Deprecated Function to coerce SummarizedExperiment to MSnSet object</td></tr>
<tr><td style="width: 25%;"><a href="test_diff.html">test_diff</a></td>
<td>Differential enrichment test</td></tr>
<tr><td style="width: 25%;"><a href="test_gsea.html">test_gsea</a></td>
<td>Gene Set Enrichment Analysis</td></tr>
<tr><td style="width: 25%;"><a href="theme_DEP1.html">theme_DEP1</a></td>
<td>DEP ggplot theme 1</td></tr>
<tr><td style="width: 25%;"><a href="theme_DEP2.html">theme_DEP2</a></td>
<td>DEP ggplot theme 2</td></tr>
<tr><td style="width: 25%;"><a href="TMT.html">TMT</a></td>
<td>TMT workflow</td></tr>
<tr><td style="width: 25%;"><a href="UbiLength.html">UbiLength</a></td>
<td>UbiLength - Ubiquitin interactors of different linear ubiquitin lengths (UbIA-MS dataset)</td></tr>
<tr><td style="width: 25%;"><a href="UbiLength_ExpDesign.html">UbiLength_ExpDesign</a></td>
<td>Experimental design of the UbiLength dataset</td></tr>
</table>
</div></body></html>
