<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Differential enrichment test</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for test_diff {DEP}"><tr><td>test_diff {DEP}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Differential enrichment test</h2>

<h3>Description</h3>

<p><code>test_diff</code> performs a differential enrichment test based on
protein-wise linear models and empirical Bayes
statistics using <span class="pkg">limma</span>. False Discovery Rates are estimated
using <span class="pkg">fdrtool</span>.
</p>


<h3>Usage</h3>

<pre>
test_diff(se, type = c("control", "all", "manual"), control = NULL,
  test = NULL, design_formula = formula(~0 + condition))
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>se</code></td>
<td>
<p>SummarizedExperiment,
Proteomics data (output from <code><a href="../../DEP/help/make_se.html">make_se</a>()</code> or
<code><a href="../../DEP/help/make_se_parse.html">make_se_parse</a>()</code>). It is adviced to first remove
proteins with too many missing values using <code><a href="../../DEP/help/filter_missval.html">filter_missval</a>()</code>,
normalize the data using <code><a href="../../DEP/help/normalize_vsn.html">normalize_vsn</a>()</code> and
impute remaining missing values using <code><a href="../../DEP/help/impute.html">impute</a>()</code>.</p>
</td></tr>
<tr valign="top"><td><code>type</code></td>
<td>
<p>&quot;control&quot;, &quot;all&quot; or &quot;manual&quot;,
The type of contrasts that will be tested.
This can be all possible pairwise comparisons (&quot;all&quot;),
limited to the comparisons versus the control (&quot;control&quot;), or
manually defined contrasts (&quot;manual&quot;).</p>
</td></tr>
<tr valign="top"><td><code>control</code></td>
<td>
<p>Character(1),
The condition to which contrasts are generated if type = &quot;control&quot;
(a control condition would be most appropriate).</p>
</td></tr>
<tr valign="top"><td><code>test</code></td>
<td>
<p>Character,
The contrasts that will be tested if type = &quot;manual&quot;.
These should be formatted as &quot;SampleA_vs_SampleB&quot; or
c(&quot;SampleA_vs_SampleC&quot;, &quot;SampleB_vs_SampleC&quot;).</p>
</td></tr>
<tr valign="top"><td><code>design_formula</code></td>
<td>
<p>Formula,
Used to create the design matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A SummarizedExperiment object
containing fdr estimates of differential expression.
</p>


<h3>Examples</h3>

<pre>
# Load example
data &lt;- UbiLength
data &lt;- data[data$Reverse != "+" &amp; data$Potential.contaminant != "+",]
data_unique &lt;- make_unique(data, "Gene.names", "Protein.IDs", delim = ";")

# Make SummarizedExperiment
columns &lt;- grep("LFQ.", colnames(data_unique))
exp_design &lt;- UbiLength_ExpDesign
se &lt;- make_se(data_unique, columns, exp_design)

# Filter, normalize and impute missing values
filt &lt;- filter_missval(se, thr = 0)
norm &lt;- normalize_vsn(filt)
imputed &lt;- impute(norm, fun = "MinProb", q = 0.01)

# Test for differentially expressed proteins
diff &lt;- test_diff(imputed, "control", "Ctrl")
diff &lt;- test_diff(imputed, "manual",
    test = c("Ubi4_vs_Ctrl", "Ubi6_vs_Ctrl"))

# Test for differentially expressed proteins with a custom design formula
diff &lt;- test_diff(imputed, "control", "Ctrl",
    design_formula = formula(~ 0 + condition + replicate))
</pre>

<hr /><div style="text-align: center;">[Package <em>DEP</em> version 1.16.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
