<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: TMT workflow</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for TMT {DEP}"><tr><td>TMT {DEP}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>TMT workflow</h2>

<h3>Description</h3>

<p><code>TMT</code> is a wrapper function running the
entire differential enrichment/expression analysis workflow
for TMT-based proteomics data.
The protein table from
<a href="http://www.nature.com/nprot/journal/v10/n10/full/nprot.2015.101.html">IsobarQuant</a>
is used as direct input.
</p>


<h3>Usage</h3>

<pre>
TMT(proteins, expdesign, fun = c("man", "bpca", "knn", "QRILC", "MLE",
  "MinDet", "MinProb", "min", "zero", "mixed", "nbavg"), type = c("all",
  "control", "manual"), control = NULL, test = NULL,
  name = "gene_name", ids = "protein_id", alpha = 0.05, lfc = 1)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>proteins</code></td>
<td>
<p>Data.frame,
The data object.</p>
</td></tr>
<tr valign="top"><td><code>expdesign</code></td>
<td>
<p>Data.frame,
The experimental design object.</p>
</td></tr>
<tr valign="top"><td><code>fun</code></td>
<td>
<p>&quot;man&quot;, &quot;bpca&quot;, &quot;knn&quot;, &quot;QRILC&quot;, &quot;MLE&quot;, &quot;MinDet&quot;,
&quot;MinProb&quot;, &quot;min&quot;, &quot;zero&quot;, &quot;mixed&quot; or &quot;nbavg&quot;,
Function used for data imputation based on <code><a href="../../DEP/help/manual_impute.html">manual_impute</a></code>
and <code><a href="../../MSnbase/help/impute-methods.html">impute</a></code>.</p>
</td></tr>
<tr valign="top"><td><code>type</code></td>
<td>
<p>'all', 'control' or 'manual',
The type of contrasts that will be generated.</p>
</td></tr>
<tr valign="top"><td><code>control</code></td>
<td>
<p>Character(1),
The sample name to which the contrasts are generated
(the control sample would be most appropriate).</p>
</td></tr>
<tr valign="top"><td><code>test</code></td>
<td>
<p>Character,
The contrasts that will be tested if type = &quot;manual&quot;.
These should be formatted as &quot;SampleA_vs_SampleB&quot; or
c(&quot;SampleA_vs_SampleC&quot;, &quot;SampleB_vs_SampleC&quot;).</p>
</td></tr>
<tr valign="top"><td><code>name</code></td>
<td>
<p>Character(1),
Name of the column representing gene names.</p>
</td></tr>
<tr valign="top"><td><code>ids</code></td>
<td>
<p>'Character(1),
Name of the column representing protein IDs.</p>
</td></tr>
<tr valign="top"><td><code>alpha</code></td>
<td>
<p>Numeric(1),
sets the false discovery rate threshold.</p>
</td></tr>
<tr valign="top"><td><code>lfc</code></td>
<td>
<p>Numeric(1),
sets the log fold change threshold.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 8 objects:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>se</code></td>
<td>
<p>SummarizedExperiment object containing the original data</p>
</td></tr>
<tr valign="top"><td><code>filt</code></td>
<td>
<p>SummarizedExperiment object containing the filtered data</p>
</td></tr>
<tr valign="top"><td><code>norm</code></td>
<td>
<p>SummarizedExperiment object containing the normalized data</p>
</td></tr>
<tr valign="top"><td><code>imputed</code></td>
<td>
<p>SummarizedExperiment object containing the imputed data</p>
</td></tr>
<tr valign="top"><td><code>diff</code></td>
<td>
<p>SummarizedExperiment object
containing FDR estimates of differential expression</p>
</td></tr>
<tr valign="top"><td><code>dep</code></td>
<td>
<p>SummarizedExperiment object
annotated with logical columns indicating significant proteins</p>
</td></tr>
<tr valign="top"><td><code>results</code></td>
<td>
<p>data.frame containing containing
all results variables from the performed analysis</p>
</td></tr>
<tr valign="top"><td><code>param</code></td>
<td>
<p>data.frame containing the test parameters</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre>
## Not run: 

TMT_res &lt;- TMT()


## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>DEP</em> version 1.16.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
