<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head><title>R: Genome wide annotation for Human</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">
<h1> Genome wide annotation for Human
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;org.Hs.eg.db&rsquo; version 3.14.0</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
</ul>

<h2>Help Pages</h2>


<table width="100%">
<tr><td style="width: 25%;"><a href="org.Hs.egBASE.html">org.Hs.eg</a></td>
<td>Bioconductor annotation data package</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egBASE.html">org.Hs.eg.db</a></td>
<td>Bioconductor annotation data package</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egACCNUM.html">org.Hs.egACCNUM</a></td>
<td>Map Entrez Gene identifiers to GenBank Accession Numbers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egACCNUM.html">org.Hs.egACCNUM2EG</a></td>
<td>Map Entrez Gene identifiers to GenBank Accession Numbers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egALIAS2EG.html">org.Hs.egALIAS2EG</a></td>
<td>Map between Common Gene Symbol Identifiers and Entrez Gene</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egCHR.html">org.Hs.egCHR</a></td>
<td>Map Entrez Gene IDs to Chromosomes</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egCHRLENGTHS.html">org.Hs.egCHRLENGTHS</a></td>
<td>A named vector for the length of each of the chromosomes</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egCHRLOC.html">org.Hs.egCHRLOC</a></td>
<td>Entrez Gene IDs to Chromosomal Location</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egCHRLOC.html">org.Hs.egCHRLOCEND</a></td>
<td>Entrez Gene IDs to Chromosomal Location</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egENSEMBL.html">org.Hs.egENSEMBL</a></td>
<td>Map Ensembl gene accession numbers with Entrez Gene identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egENSEMBL.html">org.Hs.egENSEMBL2EG</a></td>
<td>Map Ensembl gene accession numbers with Entrez Gene identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egENSEMBLPROT.html">org.Hs.egENSEMBLPROT</a></td>
<td>Map Ensembl protein acession numbers with Entrez Gene identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egENSEMBLPROT.html">org.Hs.egENSEMBLPROT2EG</a></td>
<td>Map Ensembl protein acession numbers with Entrez Gene identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egENSEMBLTRANS.html">org.Hs.egENSEMBLTRANS</a></td>
<td>Map Ensembl transcript acession numbers with Entrez Gene identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egENSEMBLTRANS.html">org.Hs.egENSEMBLTRANS2EG</a></td>
<td>Map Ensembl transcript acession numbers with Entrez Gene identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egENZYME.html">org.Hs.egENZYME</a></td>
<td>Map between Entrez Gene IDs and Enzyme Commission (EC) Numbers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egENZYME.html">org.Hs.egENZYME2EG</a></td>
<td>Map between Entrez Gene IDs and Enzyme Commission (EC) Numbers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egGENENAME.html">org.Hs.egGENENAME</a></td>
<td>Map between Entrez Gene IDs and Genes</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egGENETYPE.html">org.Hs.egGENETYPE</a></td>
<td>Map between Entrez Gene Identifiers and Gene Type</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egGO.html">org.Hs.egGO</a></td>
<td>Maps between Entrez Gene IDs and Gene Ontology (GO) IDs</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egGO.html">org.Hs.egGO2ALLEGS</a></td>
<td>Maps between Entrez Gene IDs and Gene Ontology (GO) IDs</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egGO.html">org.Hs.egGO2EG</a></td>
<td>Maps between Entrez Gene IDs and Gene Ontology (GO) IDs</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egMAP.html">org.Hs.egMAP</a></td>
<td>Map between Entrez Gene Identifiers and cytogenetic maps/bands</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egMAP.html">org.Hs.egMAP2EG</a></td>
<td>Map between Entrez Gene Identifiers and cytogenetic maps/bands</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egMAPCOUNTS.html">org.Hs.egMAPCOUNTS</a></td>
<td>Number of mapped keys for the maps in package org.Hs.eg.db</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egOMIM.html">org.Hs.egOMIM</a></td>
<td>Map between Entrez Gene Identifiers and Mendelian Inheritance in Man (MIM) identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egOMIM.html">org.Hs.egOMIM2EG</a></td>
<td>Map between Entrez Gene Identifiers and Mendelian Inheritance in Man (MIM) identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egORGANISM.html">org.Hs.egORGANISM</a></td>
<td>The Organism for org.Hs.eg</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egPATH.html">org.Hs.egPATH</a></td>
<td>Mappings between Entrez Gene identifiers and KEGG pathway identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egPATH.html">org.Hs.egPATH2EG</a></td>
<td>Mappings between Entrez Gene identifiers and KEGG pathway identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egPFAM.html">org.Hs.egPFAM</a></td>
<td>Maps between Manufacturer Identifiers and PFAM Identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egPMID.html">org.Hs.egPMID</a></td>
<td>Map between Entrez Gene Identifiers and PubMed Identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egPMID.html">org.Hs.egPMID2EG</a></td>
<td>Map between Entrez Gene Identifiers and PubMed Identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egPROSITE.html">org.Hs.egPROSITE</a></td>
<td>Maps between Manufacturer Identifiers and PROSITE Identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egREFSEQ.html">org.Hs.egREFSEQ</a></td>
<td>Map between Entrez Gene Identifiers and RefSeq Identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egREFSEQ.html">org.Hs.egREFSEQ2EG</a></td>
<td>Map between Entrez Gene Identifiers and RefSeq Identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egSYMBOL.html">org.Hs.egSYMBOL</a></td>
<td>Map between Entrez Gene Identifiers and Gene Symbols</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egSYMBOL.html">org.Hs.egSYMBOL2EG</a></td>
<td>Map between Entrez Gene Identifiers and Gene Symbols</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egUCSCKG.html">org.Hs.egUCSCKG</a></td>
<td>This mapping has been deprecated and will no longer be available after bioconductor 2.6.  See the details section for how you can live without it.  For now, it is a map of UCSC "Known Gene" accession numbers with Entrez Gene identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.egUNIPROT.html">org.Hs.egUNIPROT</a></td>
<td>Map Uniprot accession numbers with Entrez Gene identifiers</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.eg_dbconn.html">org.Hs.eg_dbconn</a></td>
<td>Collect information about the package annotation DB</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.eg_dbconn.html">org.Hs.eg_dbfile</a></td>
<td>Collect information about the package annotation DB</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.eg_dbconn.html">org.Hs.eg_dbInfo</a></td>
<td>Collect information about the package annotation DB</td></tr>
<tr><td style="width: 25%;"><a href="org.Hs.eg_dbconn.html">org.Hs.eg_dbschema</a></td>
<td>Collect information about the package annotation DB</td></tr>
</table>
</div></body></html>
