<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: This mapping has been deprecated and will no longer be...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for org.Hs.egUCSCKG {org.Hs.eg.db}"><tr><td>org.Hs.egUCSCKG {org.Hs.eg.db}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>This mapping has been deprecated and will no longer be available
after bioconductor 2.6.  See the details section for how you can live
without it.  For now, it is a map of UCSC &quot;Known Gene&quot; accession numbers with Entrez Gene identifiers</h2>

<h3>Description</h3>

 
<p>org.Hs.egUCSCKG is an R object that contains mappings between 
Entrez Gene identifiers and UCSC &quot;Known Gene&quot; accession numbers.    
</p>


<h3>Details</h3>

<p>This object is a simple mapping of Entrez Gene identifiers
<a href="https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gene">https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gene</a> to UCSC
&quot;Known Gene&quot; Accession Numbers.  Please be advised that UCSC &quot;Known
Genes&quot; are not more like putative transcripts transcript identifiers
than they are gene identifiers.  
</p>
<p>Mappings were based on data provided by: UCSC Genome Bioinformatics (Homo sapiens)  
</p>
<p>With a date stamp from the source of: 2021-Jul20
</p>
<p>This mapping is based on the very latest build available at UCSC for
this organism as of March 2010.  2.6 is the last release where you
can expect it to be here.  The GenomicFeatures package contains
functionality that replaces the need for this mapping.  Please see the
GenomicFeatures vignette for details on how to use it.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="../../AnnotationDbi/help/AnnotationDb-class.html">AnnotationDb-class</a></code> for use of
the <code>select()</code> interface.
</p>
</li></ul>



<h3>Examples</h3>

<pre>
## select() interface:
## Objects in this package can be accessed using the select() interface
## from the AnnotationDbi package. See ?select for details.

## Bimap interface:
x &lt;- org.Hs.egUCSCKG
# Get the entrez gene IDs that are mapped to a UCSC KG ID
mapped_genes &lt;- mappedkeys(x)
# Convert to a list
xx &lt;- as.list(x[mapped_genes])
if(length(xx) &gt; 0) {
  # Get the UCSC gene IDs for the first five genes
  xx[1:5]
  # Get the first one
  xx[[1]]
}
</pre>

<hr /><div style="text-align: center;">[Package <em>org.Hs.eg.db</em> version 3.14.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
