<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Map between Entrez Gene Identifiers and RefSeq Identifiers</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for org.Hs.egREFSEQ {org.Hs.eg.db}"><tr><td>org.Hs.egREFSEQ {org.Hs.eg.db}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Map between Entrez Gene Identifiers and RefSeq Identifiers</h2>

<h3>Description</h3>

<p>org.Hs.egREFSEQ is an R object that provides mappings between
entrez gene identifiers and RefSeq identifiers.
</p>


<h3>Details</h3>

<p>Each entrez gene identifier is mapped to a named vector of RefSeq
identifiers. The name represents the entrez gene identifier and the
vector contains all RefSeq identifiers that can be mapped to that
entrez gene identifier. The length of the vector may be one or
greater, depending on how many RefSeq identifiers a given entrez gene
identifier can be mapped to. An <code>NA</code> is reported for any entrex
gene identifier that cannot be mapped to a RefSeq identifier at this
time.
</p>
<p>RefSeq identifiers differ in format according to the type of record the identifiers
are for as shown below:
</p>
<p>NG\_XXXXX: RefSeq accessions for genomic region (nucleotide) records
</p>
<p>NM\_XXXXX: RefSeq accessions for mRNA records
</p>
<p>NC\_XXXXX: RefSeq accessions for chromosome records
</p>
<p>NP\_XXXXX: RefSeq accessions for protein records
</p>
<p>XR\_XXXXX: RefSeq accessions for model RNAs that are not associated with
protein products
</p>
<p>XM\_XXXXX: RefSeq accessions for model mRNA records
</p>
<p>XP\_XXXXX: RefSeq accessions for model protein records
</p>
<p>Where XXXXX is a sequence of integers.
</p>
<p>NCBI <a href="https://www.ncbi.nlm.nih.gov/RefSeq/">https://www.ncbi.nlm.nih.gov/RefSeq/</a> allows users to query
the RefSeq database using RefSeq identifiers.
</p>
<p>Mappings were based on data provided by: Entrez Gene  
ftp://ftp.ncbi.nlm.nih.gov/gene/DATA  
With a date stamp from the source of: 2021-Sep13
</p>


<h3>References</h3>

<p><a href="https://www.ncbi.nlm.nih.gov">https://www.ncbi.nlm.nih.gov</a>
<a href="https://www.ncbi.nlm.nih.gov/RefSeq/">https://www.ncbi.nlm.nih.gov/RefSeq/</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="../../AnnotationDbi/help/AnnotationDb-class.html">AnnotationDb-class</a></code> for use of
the <code>select()</code> interface.
</p>
</li></ul>



<h3>Examples</h3>

<pre>
## select() interface:
## Objects in this package can be accessed using the select() interface
## from the AnnotationDbi package. See ?select for details.

## Bimap interface:
x &lt;- org.Hs.egREFSEQ
# Get the entrez gene identifiers that are mapped to any RefSeq ID
mapped_genes &lt;- mappedkeys(x)
# Convert to a list
xx &lt;- as.list(x[mapped_genes])
if(length(xx) &gt; 0) {
  # Get the REFSEQ for the first five genes
  xx[1:5]
  # Get the first one
  xx[[1]]
}

# For the reverse map:
x &lt;- org.Hs.egREFSEQ2EG
# Get the RefSeq identifier that are mapped to an entrez gene ID
mapped_seqs &lt;- mappedkeys(x)
# Convert to a list
xx &lt;- as.list(x[mapped_seqs])
if(length(xx) &gt; 0) {
  # Get the entrez gene for the first five Refseqs
  xx[1:5]
  # Get the first one
  xx[[1]]
}
</pre>

<hr /><div style="text-align: center;">[Package <em>org.Hs.eg.db</em> version 3.14.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
