<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Entrez Gene IDs to Chromosomal Location</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for org.Hs.egCHRLOC {org.Hs.eg.db}"><tr><td>org.Hs.egCHRLOC {org.Hs.eg.db}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Entrez Gene IDs to Chromosomal Location</h2>

<h3>Description</h3>

<p>org.Hs.egCHRLOC is an R object that maps entrez gene identifiers to the
starting position of the gene. The position of a gene is
measured as the number of base pairs.
</p>
<p>The CHRLOCEND mapping is the same as the CHRLOC mapping except that it
specifies the ending base of a gene instead of the start.
</p>


<h3>Details</h3>

<p>Each entrez gene identifier maps to a named vector of chromosomal locations,
where the name indicates the chromosome. 
</p>
<p>Chromosomal locations on both the sense and antisense strands are
measured as the number of base pairs from the p (5' end of the sense
strand) to q (3' end of the sense strand) arms. Chromosomal locations on the
antisense strand have a leading &quot;-&quot; sign (e. g. -1234567). 
</p>
<p>Since some genes have multiple start sites, this field can map to
multiple locations.
</p>
<p>Mappings were based on data provided by: UCSC Genome Bioinformatics (Homo sapiens)  
</p>
<p>With a date stamp from the source of: 2021-Jul20
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="../../AnnotationDbi/help/AnnotationDb-class.html">AnnotationDb-class</a></code> for use of
the <code>select()</code> interface.
</p>
</li></ul>



<h3>Examples</h3>

<pre>
## select() interface:
## Objects in this package can be accessed using the select() interface
## from the AnnotationDbi package. See ?select for details.

## Bimap interface:
x &lt;- org.Hs.egCHRLOC
# Get the entrez gene identifiers that are mapped to chromosome locations
mapped_genes &lt;- mappedkeys(x)
# Convert to a list
xx &lt;- as.list(x[mapped_genes])
if(length(xx) &gt; 0) {
  # Get the CHRLOC for the first five genes
  xx[1:5]
  # Get the first one
  xx[[1]]
}
</pre>

<hr /><div style="text-align: center;">[Package <em>org.Hs.eg.db</em> version 3.14.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
