<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Fit the vsn model</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for vsn2 {vsn}"><tr><td>vsn2 {vsn}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Fit the vsn model</h2>

<h3>Description</h3>

<p><code>vsn2</code> fits the vsn model to the data 
in <code>x</code> and returns a <code><a href="../../vsn/help/vsn-class.html">vsn</a></code> object with
the fit parameters and the transformed data matrix.
The data are, typically, feature intensity readings from a
microarray, but this function may also be useful for other kinds of
intensity data that obey an additive-multiplicative error model.
To obtain an object of the same class as <code>x</code>, containing
the normalised data and the same metdata as <code>x</code>, use
</p>
<pre>
    fit = vsn2(x, ...)
    nx = predict(fit, newdata=x)
  </pre>
<p>or the wrapper <code><a href="../../vsn/help/justvsn.html">justvsn</a></code>.
Please see the vignette <em>Introduction to vsn</em>.
</p>


<h3>Usage</h3>

<pre>
vsnMatrix(x,
          reference,
          strata,
          lts.quantile = 0.9,
          subsample    = 0L,
          verbose      = interactive(),
          returnData   = TRUE,
          calib        = "affine",
          pstart,
          minDataPointsPerStratum = 42L,
          optimpar     = list(),
          defaultpar   = list(factr=5e7, pgtol=2e-4, maxit=60000L,
                              trace=0L, cvg.niter=7L, cvg.eps=0))

## S4 method for signature 'ExpressionSet'
vsn2(x, reference, strata, ...)

## S4 method for signature 'AffyBatch'
vsn2(x, reference, strata, subsample, ...)

## S4 method for signature 'NChannelSet'
vsn2(x, reference, strata, backgroundsubtract=FALSE,
       foreground=c("R","G"), background=c("Rb", "Gb"), ...)

## S4 method for signature 'RGList'
vsn2(x, reference, strata, ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>x</code></td>
<td>
<p>An object containing the data to which the model is fitted.</p>
</td></tr>
<tr valign="top"><td><code>reference</code></td>
<td>
<p>Optional, a <code><a href="../../vsn/help/vsn-class.html">vsn</a></code> object from
a previous fit. If this argument is specified, the data in <code>x</code>
are normalized &quot;towards&quot; an existing set of reference arrays whose
parameters are stored in the object <code>reference</code>. If this
argument is not specified, then the data in <code>x</code> are normalized
&quot;among themselves&quot;. See Details for a more precise explanation.</p>
</td></tr>
<tr valign="top"><td><code>strata</code></td>
<td>
<p>Optional, a <code>factor</code> or <code>integer</code>
whose length is <code>nrow(x)</code>. It can
be used for stratified normalization (i.e. separate offsets <i>a</i> and
factors <i>b</i> for each level of <code>strata</code>). If missing, all
rows of <code>x</code> are assumed to come from one stratum.
If <code>strata</code> is an integer, its values must cover the range
<i>1,&hellip;,n</i>, where <i>n</i> is the number of strata.</p>
</td></tr>
<tr valign="top"><td><code>lts.quantile</code></td>
<td>
<p>Numeric of length 1. The quantile that is used for
the resistant least trimmed sum of squares regression. Allowed
values are between 0.5 and 1. A value of 1 corresponds to ordinary
least sum of squares regression.</p>
</td></tr>
<tr valign="top"><td><code>subsample</code></td>
<td>
<p>Integer of length 1. If its value is greater than 0,
the model parameters are
estimated from a subsample of the data of size <code>subsample</code>
only, yet the fitted transformation is
then applied to all data. For large datasets, this can substantially
reduce the CPU time and memory consumption at a negligible loss of
precision. Note that the <code>AffyBatch</code> method of <code>vsn2</code> sets a value of
<code>30000</code> for this parameter if it is missing from the function
call - which is different from the behaviour of the other methods.</p>
</td></tr>
<tr valign="top"><td><code>backgroundsubtract</code></td>
<td>
<p>Logical of length 1: should
local background estimates be subtracted before fitting vsn?</p>
</td></tr>
<tr valign="top"><td><code>foreground, background</code></td>
<td>
<p>Aligned character vectors of the same length,
naming the channels of <code>x</code> that should be used
as foreground and background values.</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>Logical. If TRUE, some messages are printed.</p>
</td></tr>
<tr valign="top"><td><code>returnData</code></td>
<td>
<p>Logical. If TRUE, the transformed data are returned
in a slot of the resulting <code><a href="../../vsn/help/vsn-class.html">vsn</a></code> object.
Setting this option to <code>FALSE</code> allows saving memory
if the data are not needed.</p>
</td></tr>
<tr valign="top"><td><code>calib</code></td>
<td>
<p>Character of length 1. Allowed values are <code>affine</code>
and <code>none</code>. The default, <code>affine</code>, corresponds to the
behaviour in package versions &lt;= 3.9, and to what is described in
references [1] and [2]. The option <code>none</code> is an experimental
new feature, in which no affine calibration is performed and only
two global variance stabilisation transformation parameters <code>a</code>
and <code>b</code> are fitted. This functionality might be useful in
conjunction with other calibration methods, such as quantile
normalisation - see the vignette <em>Introduction to vsn</em>.</p>
</td></tr>
<tr valign="top"><td><code>pstart</code></td>
<td>
<p>Optional, a three-dimensional numeric array that
specifies start values for the iterative parameter
estimation algorithm.
If not specified, the function tries to guess useful start values.
The first dimension corresponds to the levels of <code>strata</code>,
the second dimension to the columns of <code>x</code> and the third dimension
must be 2, corresponding to offsets and factors.</p>
</td></tr>
<tr valign="top"><td><code>minDataPointsPerStratum</code></td>
<td>
<p>The minimum number of data points per
stratum. Normally there is no need for the user to change this;
refer to the vignette for further documentation.</p>
</td></tr>
<tr valign="top"><td><code>optimpar</code></td>
<td>
<p>Optional, a list with parameters for the likelihood
optimisation algorithm. Default parameters are taken from
<code>defaultpar</code>. See details.</p>
</td></tr>
<tr valign="top"><td><code>defaultpar</code></td>
<td>
<p>The default parameters  for the likelihood
optimisation algorithm. Values in <code>optimpar</code> take precedence
over those in <code>defaultpar</code>. The purpose of this argument is to
expose the default values in this manual page - it is not
intended to be changed, please use <code>optimpar</code> for that.</p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>Arguments that get passed on to <code>vsnMatrix</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="../../vsn/help/vsn-class.html">vsn</a></code>.</p>


<h3>Note on overall scale and location of the glog
transformation</h3>

<p>The data are returned on a <i>glog</i> scale to base 2. More precisely,
the transformed data are subject to the transformation
<i>glog_2(f(b)*x+a) + c</i>, where the function
<i>glog_2(u) = log_2(u+&radic;{u*u+1}) = asinh(u)/\log(2)</i> is called the
generalised logarithm, the offset <i>a</i> and the scaling parameter
<i>b</i> are the fitted model parameters
(see references), and <i>f(x)=\exp(x)</i> is a parameter transformation that
allows ensuring positivity of the factor in front of <i>x</i> while
using an unconstrained optimisation over <i>b</i> [4].
The overall offset <i>c</i> is computed from the <i>b</i>'s such that for
large <i>x</i> the transformation approximately corresponds to the
<i>\log_2</i> function. This is done separately for each stratum, but with the
same value across arrays. More precisely, if the element <code>b[s,i]</code>
of the array <em>b</em> is the scaling parameter for the <code>s</code>-th
stratum and the <code>i</code>-th array, then <code>c[s]</code> is computed as
<code>log2(2*f(mean(b[,i])))</code>.
The offset <em>c</em> is inconsequential for all differential
expression calculations, but many users like to see the data in a
range that they are familiar with.
</p>


<h3>Specific behaviour of the different methods</h3>

<p><code>vsn2</code> methods exist for
<code><a href="../../Biobase/help/class.ExpressionSet.html">ExpressionSet</a></code>,
<code><a href="../../Biobase/help/class.NChannelSet.html">NChannelSet</a></code>,
<code><a href="../../affy/help/AffyBatch-class.html">AffyBatch</a></code> (from the <code>affy</code> package),
<code><a href="../../limma/help/rglist.html">RGList</a></code> (from the <code>limma</code> package),
<code>matrix</code> and <code>numeric</code>.
If <code>x</code> is an <code><a href="../../Biobase/help/class.NChannelSet.html">NChannelSet</a></code>, then
<code>vsn2</code> is applied to the matrix that is obtained
by horizontally concatenating the color channels.
Optionally, available background estimates can be subtracted before.
If <code>x</code> is an <code><a href="../../limma/help/rglist.html">RGList</a></code>, it is
converted into an <code>NChannelSet</code>
using a copy of Martin Morgan's code for <code>RGList</code> to
<code>NChannelSet</code> coercion, then the <code>NChannelSet</code> method is called.
</p>


<h3>Standalone versus reference normalisation</h3>

<p>If the <code>reference</code> argument is <em>not</em> specified, then the model
parameters <i>&mu;_k</i> and <i>&sigma;</i> are fit from the data in <code>x</code>.
This is the mode of operation described in [1]
and that was the only option in versions 1.X of this package.
If <code>reference</code> is specified, the model parameters
<i>&mu;_k</i> and <i>&sigma;</i> are taken from it.
This allows for 'incremental' normalization [4].</p>


<h3>Convergence of the iterative likelihood optimisation</h3>

<p><code>L-BFGS-B</code> uses three termination criteria:
</p>

<ol>
<li> <p><code>(f_k - f_{k+1}) / max(|f_k|, |f_{k+1}|, 1) &lt;= factr * epsmch</code>
where <code>epsmch</code> is the machine precision.
</p>
</li>
<li> <p><code>|gradient| &lt; pgtol</code>
</p>
</li>
<li> <p><code>iterations &gt; maxit</code>
</p>
</li></ol>

<p>These are set by the elements <code>factr</code>, <code>pgtol</code> and
<code>maxit</code> of <code>optimpar</code>. The remaining elements are
</p>

<dl>
<dt><code>trace</code></dt><dd><p>An integer between 0 and 6, indicating the
verbosity level of <code>L-BFGS-B</code>, higher values
create more output.</p>
</dd>
<dt><code>cvg.niter</code></dt><dd><p>The number of iterations to be used in the least
trimmed sum of squares regression.</p>
</dd>
<dt><code>cvg.eps</code></dt><dd><p>Numeric. A convergence threshold for the least
trimmed sum of squares regression.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Wolfgang Huber</p>


<h3>References</h3>

<p>[1] Variance stabilization applied to microarray data
calibration and to the quantification of differential expression,
Wolfgang Huber, Anja von Heydebreck, Holger Sueltmann, Annemarie
Poustka, Martin Vingron; Bioinformatics (2002) 18 Suppl.1 S96-S104.
</p>
<p>[2] Parameter estimation for the calibration and variance stabilization 
of microarray data, 
Wolfgang Huber, Anja von Heydebreck, Holger Sueltmann, 
Annemarie Poustka, and Martin Vingron;  
Statistical Applications in Genetics and Molecular Biology (2003)
Vol. 2 No. 1, Article 3.
http://www.bepress.com/sagmb/vol2/iss1/art3.
</p>
<p>[3] L-BFGS-B: Fortran Subroutines for Large-Scale Bound Constrained
Optimization, C. Zhu, R.H. Byrd, P. Lu and J. Nocedal,
Technical Report, Northwestern University (1996).
</p>
<p>[4] Package vignette: Likelihood Calculations for vsn
</p>


<h3>See Also</h3>

<p><code><a href="../../vsn/help/justvsn.html">justvsn</a></code>, <code><a href="../help/vsn2trsf.html">predict</a></code></p>


<h3>Examples</h3>

<pre>
data("kidney")

fit = vsn2(kidney)                   ## fit
nkid = predict(fit, newdata=kidney)  ## apply fit

plot(exprs(nkid), pch=".")
abline(a=0, b=1, col="red")
</pre>

<hr /><div style="text-align: center;">[Package <em>vsn</em> version 3.62.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
